### ISSUE TITLE
App crashes on startup in Android when react-native-plaid-link-sdk is included

### ISSUE THREAD

**Issue by drekembe** [plaid/react-native-plaid-link-sdk#300]
# The problem

When I include and set up react-native-plaid-link-sdk in my react-native project, the app crashes on Android on startup as soon as it finishes loading data from the bundler.

I followed the instructions in the 6.0.4 readme to set it up. When I remove the `implementation project(':react-native-plaid-link-sdk')` line from the `app/build.gradle` file and the `packages.add(new PlaidPackage());` line (along with the import) from `MainApplication.java`, the crash no longer happens.

This is an issue that stems from me trying to resolve #296 and following the kind suggestions of @JeroenMols 

I made a separate issue for this, as #269 was a build time error for an old version of the SDK, while this is a runtime crash for a relatively new version.

# Environment

|                        |  |
| ------------------------| ---------- |
| Plaid Link React Native          | 6.0.4                |
| ReactNative Version     |  0.61.5                |
| Occurs on Android             |  yes                |
| Occurs on iOS     | no                |

Tried it on Nexus6 emulator running Android 9 with API v.28, Nexus6 emulator running Android 10 with API v.29, and Xiaomi A2 physical device running Android 10 with API v.29.

# Steps to Reproduce

I have created a minimal reproduction repo for the issue on this branch: https://github.com/drekembe/rnExamplePlaidAndroid/tree/rn0.61.5-plaid6.0.4

I encounter this crash on our real project as well as this example one.

To recreate, just do `yarn install` and then `yarn android`. If you remove plaid from `app/build.gradle` and `MainApplication.java`, the app runs fine.

# Logs

 I tried using logcat to pinpoint why the crash happens. It didn't provide any clear info, and this seems like the most relevant log output, which is emitted before the crash:

```
[   94.508041] type=1400 audit(1610021676.312:20): avc: denied { write } for comm="RenderThread" name="property_service" dev="tmpfs" ino=824 scontext=u:r:untrusted_app_27:s0:c134,c256,c512,c768 tcontext=u:object_r:property_socket:s0 tclass=sock_file permissive=0 app=com.rnexampleplaidandroid
[   95.354445] binder: undelivered TRANSACTION_COMPLETE
[   95.356588] binder: undelivered transaction 145726, process died.
```

I'm not very experienced in debugging android crashes, so this is the best I could come up with regarding debug info for the crash.

**Comment by JeroenMols** [comment]
Thanks for reporting and for the sample project!

I'm getting the following error after running your sample:

```
2021-01-11 15:04:22.382 5664-5711/com.rnexampleplaidandroid E/AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
    Process: com.rnexampleplaidandroid, PID: 5664
    java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/swiperefreshlayout/widget/SwipeRefreshLayout;
        at com.facebook.react.shell.MainReactPackage.createViewManagers(MainReactPackage.java:185)
        at com.facebook.react.ReactInstanceManager.getOrCreateViewManagers(ReactInstanceManager.java:782)
        at com.facebook.react.CoreModulesPackage.createUIManager(CoreModulesPackage.java:184)
        at com.facebook.react.CoreModulesPackage.getModule(CoreModulesPackage.java:152)
        at com.facebook.react.TurboReactPackage$ModuleHolderProvider.get(TurboReactPackage.java:122)
        at com.facebook.react.TurboReactPackage$ModuleHolderProvider.get(TurboReactPackage.java:110)
        at com.facebook.react.bridge.ModuleHolder.create(ModuleHolder.java:188)
        at com.facebook.react.bridge.ModuleHolder.getModule(ModuleHolder.java:153)
        at com.facebook.react.bridge.NativeModuleRegistry.getModule(NativeModuleRegistry.java:148)
        at com.facebook.react.bridge.CatalystInstanceImpl.getNativeModule(CatalystInstanceImpl.java:479)
        at com.facebook.react.bridge.CatalystInstanceImpl.getNativeModule(CatalystInstanceImpl.java:466)
        at com.facebook.react.uimanager.UIManagerHelper.getUIManager(UIManagerHelper.java:31)
        at com.facebook.react.ReactInstanceManager.attachRootViewToInstance(ReactInstanceManager.java:1054)
        at com.facebook.react.ReactInstanceManager.setupReactContext(ReactInstanceManager.java:1012)
        at com.facebook.react.ReactInstanceManager.access$1400(ReactInstanceManager.java:125)
        at com.facebook.react.ReactInstanceManager$5$2.run(ReactInstanceManager.java:972)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:26)
        at android.os.Looper.loop(Looper.java:223)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:225)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "androidx.swiperefreshlayout.widget.SwipeRefreshLayout" on path: DexPathList[[zip file "/data/app/~~RdkaCRCxMVkQCGQTBLKQNw==/com.rnexampleplaidandroid-Fw2u12mp1PCjbSwsPUaN0g==/base.apk"],nativeLibraryDirectories=[/data/app/~~RdkaCRCxMVkQCGQTBLKQNw==/com.rnexampleplaidandroid-Fw2u12mp1PCjbSwsPUaN0g==/lib/x86, /data/app/~~RdkaCRCxMVkQCGQTBLKQNw==/com.rnexampleplaidandroid-Fw2u12mp1PCjbSwsPUaN0g==/base.apk!/lib/x86, /system/lib, /system_ext/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:207)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.facebook.react.shell.MainReactPackage.createViewManagers(MainReactPackage.java:185) 
        at com.facebook.react.ReactInstanceManager.getOrCreateViewManagers(ReactInstanceManager.java:782) 
        at com.facebook.react.CoreModulesPackage.createUIManager(CoreModulesPackage.java:184) 
        at com.facebook.react.CoreModulesPackage.getModule(CoreModulesPackage.java:152) 
        at com.facebook.react.TurboReactPackage$ModuleHolderProvider.get(TurboReactPackage.java:122) 
        at com.facebook.react.TurboReactPackage$ModuleHolderProvider.get(TurboReactPackage.java:110) 
        at com.facebook.react.bridge.ModuleHolder.create(ModuleHolder.java:188) 
        at com.facebook.react.bridge.ModuleHolder.getModule(ModuleHolder.java:153) 
        at com.facebook.react.bridge.NativeModuleRegistry.getModule(NativeModuleRegistry.java:148) 
        at com.facebook.react.bridge.CatalystInstanceImpl.getNativeModule(CatalystInstanceImpl.java:479) 
        at com.facebook.react.bridge.CatalystInstanceImpl.getNativeModule(CatalystInstanceImpl.java:466) 
        at com.facebook.react.uimanager.UIManagerHelper.getUIManager(UIManagerHelper.java:31) 
        at com.facebook.react.ReactInstanceManager.attachRootViewToInstance(ReactInstanceManager.java:1054) 
        at com.facebook.react.ReactInstanceManager.setupReactContext(ReactInstanceManager.java:1012) 
        at com.facebook.react.ReactInstanceManager.access$1400(ReactInstanceManager.java:125) 
        at com.facebook.react.ReactInstanceManager$5$2.run(ReactInstanceManager.java:972) 
        at android.os.Handler.handleCallback(Handler.java:938) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:26) 
        at android.os.Looper.loop(Looper.java:223) 
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:225) 
        at java.lang.Thread.run(Thread.java:923) 
 ```

Which I've solved by adding the following dependencies to `android/app/build.gradle`:

```groovy
dependencies {
    ....
    implementation 'androidx.appcompat:appcompat:1.2.0'
    implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0'
}
```

Can you please try if this solution also works for you?

**Comment by drekembe** [comment]
I tried it out and it works! I had to migrate to the new link token flow, but we'd have to do that sooner or later anyway. It also works on RN 0.60.6, for what it's worth. Thanks for all your help.

One more thing, how did you get the Java stack trace? I tried getting crash logs by running the emulator with `emulator -no-snapshot-load -avd Nexus6 -logcat -v`, but never got anything resembling a Java stack trace in those logs.
